apps/web/src/hooks/useSafeTokenAllocation.ts

```diff
@@ -43,49 +42,42 @@ const tokenInterface = new Interface(['function balanceOf(address _owner) public
   'function getUserTokenBalance(address holder) external view returns (uint96 amount)',
 ])
-
-export const _getRedeemDeadline = memoize(
-  async (allocation: VestingData, web3ReadOnly: JsonRpcProvider): Promise<string> => {
-    return web3ReadOnly.call({
-      to: allocation.contract,
-      data: airdropInterface.encodeFunctionData('redeemDeadline'),
-    })
-  },
-  ({ chainId, contract }) => chainId + contract,
-)
-
 /**
  * Add on-chain information to allocation.
  * Fetches if the redeem deadline is expired and the claimed tokens from on-chain
  */
-const completeAllocation = async (allocation: VestingData): Promise<Vesting> => {
+const completeAllocations = async (allocations: VestingData[]): Promise<Vesting[]> => {
   const web3ReadOnly = getWeb3ReadOnly()
   if (!web3ReadOnly) {
-    throw new Error('Cannot fetch vesting without web3 provider')
+    throw new Error('Cannot fetch vestings without web3 provider')
   }
-  const onChainVestingData = await web3ReadOnly.call({
+
+  const calls = allocations.map((allocation) => ({
     to: allocation.contract,
     data: airdropInterface.encodeFunctionData('vestings', [allocation.vestingId]),
-  })
+  }))
+  const results = await multicall(web3ReadOnly, calls)
 
-  const decodedVestingData = AbiCoder.defaultAbiCoder().decode(
-    // account, curveType, managed, durationWeeks, startDate, amount, amountClaimed, pausingDate, cancelled}
-    ['address', 'uint8', 'bool', 'uint16', 'uint64', 'uint128', 'uint128', 'uint64', 'bool'],
-    onChainVestingData,
-  )
-
-  const isRedeemed = decodedVestingData[0].toLowerCase() !== ZERO_ADDRESS.toLowerCase()
-  if (isRedeemed) {
-    return { ...allocation, isRedeemed, isExpired: false, amountClaimed: decodedVestingData[6] }
-  }
+  return allocations.map((allocation, index) => {
+    const result = results[index]
+    if (!result.success) {
+      throw new Error(`Failed to fetch vesting data for ${allocation.vestingId}`)
+    }
 
-  // Allocation is not yet redeemed => check the redeemDeadline
-  const redeemDeadline = await _getRedeemDeadline(allocation, web3ReadOnly)
+    const decodedVestingData = AbiCoder.defaultAbiCoder().decode(
+      // account, curveType, managed, durationWeeks, startDate, amount, amountClaimed, pausingDate, cancelled}
+      ['address', 'uint8', 'bool', 'uint16', 'uint64', 'uint128', 'uint128', 'uint64', 'bool'],
+      result.returnData,
+    )
 
-  const redeemDeadlineDate = new Date(Number(BigInt(redeemDeadline) * BigInt(1000)))
+    const isRedeemed = decodedVestingData[0].toLowerCase() !== ZERO_ADDRESS.toLowerCase()
+    if (isRedeemed) {
+      return { ...allocation, isRedeemed, isExpired: false, amountClaimed: decodedVestingData[6] }
+    }
 
-  // Allocation is valid if redeem deadline is in future
-  return { ...allocation, isRedeemed, isExpired: isPast(redeemDeadlineDate), amountClaimed: '0' }
+    // All allocations are expired by now. We do not load the redeem deadline anymore
```

Comment:
What about the SAP airdrop that can be redeemed until end of this year?

Reply:
@usame-algan whilst re-introducing the redeemDeadline check I realized that we only use these allocations to compute the total voting power of the current Safe and the SAP airdrop allocation does not count towards that voting power (until claimed). For this reason it does not make a different if we return it or filter it out as expired.

We can leave it as is for this reason and save these RPC requests.